apply-roles

```diff
@@ -46,11 +42,24 @@ create_custom_role() {
       ], \
       \"DataActions\": [], \
       \"NotDataActions\": [], \
-      \"AssignableScopes\":[${assignable_scopes}] \
-    }" >/dev/null
+      \"AssignableScopes\": [${assignable_scopes}] \
+    }"
+}
+
+create_custom_role() {
+  local subscription_ids_in_active_directory=("$@")
+  local attempts_to_retrieve_role=0
+  local role_definition
+
+  echo " - Creating custom role definition..."
+
+  role_definition=$(generate_custom_role_definition "${subscription_ids_in_active_directory[@]}")
+
+  az role definition create \
+    --role-definition "${role_definition}" >/dev/null
 
   # It can take a bit of time for the role to be available to use so query until it is available
-  while [[ -z "${custom_role_definition_id}" ]] && [[ attempts_to_retrieve_role -lt 5 ]]; do
+  while [[ -z "${custom_role_definition_id}" ]] && [[ attempts_to_retrieve_role -lt 8 ]]; do
```

Comment:
I increased this to 8 as one time 5 wasn't enough so don't want the script to fail for no good reason so I increased it.